src/Components/Pagination/Pagination.tsx

```diff
@@ -1,52 +1,44 @@
-import { useEffect } from 'react';
-import classes from './Pagination.module.css';
-import { PaginationProps } from './types';
-import { useSearchParams } from 'react-router-dom';
-import { useAppDispatch, useAppSelector } from '../../store/hooks';
+import { useRouter } from 'next/router';
+import { store } from '../../store';
 import { updateCurrentPage } from '../../store/searchSlice';
+import classes from './Pagination.module.css';
 
-const Pagination = (props: PaginationProps) => {
-  const [searchParams, setSearchParams] = useSearchParams();
-  const pageCount = useAppSelector((state) => state.search.pageCount);
-  const currentPage = useAppSelector((state) => state.search.currentPage);
-
-  const dispatch = useAppDispatch();
+export default function Pagination() {
+  const router = useRouter();
+  const { page, search } = router.query;
+  const currentPage = (page || store.getState().search.currentPage) as number;
+  const pageCount = store.getState().search.pageCount;
 
-  useEffect(() => {
-    if (searchParams.get('page')) {
-      dispatch(updateCurrentPage(Number(searchParams.get('page'))));
-    } else {
-      dispatch(updateCurrentPage(1));
-    }
-  }, [searchParams, dispatch]);
+  if (page) {
+    store.dispatch(updateCurrentPage(page));
+  }
 
   function setNextPage() {
-    console.log(currentPage);
-    dispatch(updateCurrentPage(currentPage + 1));
-    setSearchParams({
-      search: localStorage.getItem('searchValue') || '',
-      page: (currentPage + 1).toString(),
+    router.push({
+      pathname: '/',
+      query: { page: Number(currentPage) + 1, search },
     });
+    store.dispatch(updateCurrentPage(Number(currentPage) + 1));
   }
 
   function setPreviousPage() {
-    dispatch(updateCurrentPage(currentPage - 1));
-    setSearchParams({
-      search: localStorage.getItem('searchValue') || '',
-      page: (currentPage - 1).toString(),
+    router.push({
+      pathname: '/',
+      query: { page: currentPage - 1, search },
     });
+    store.dispatch(updateCurrentPage(Number(currentPage) - 1));
```

Comment:
why you use convertation  `Number(currentPage) - 1` here? but don't use it above `currentPage - 1`